docs/rfc/rfc-5.adoc

```diff
@@ -0,0 +1,121 @@
+:toc: macro
+
+= RFC 5: Emptying and closing wallets
+
+:icons: font
+:numbered:
+toc::[]
+
+== Overview
+
+If a wallet still containing BTC is to be closed,
+the remaining BTC must be transferred to a new wallet if possible.
+
+This applies in the following situations:
+
+- a wallet has fallen below a specified minimum BTC amount
+- a wallet has failed a heartbeat due to losing too many operators
+- a wallet has reached a specified maximum lifetime
```

Comment:
...which should always be shorter than the undelegation time, right? 

Say we are at C_h + 1 and one whale undelegates and that whale was lucky enough to have more than C_h - T members in the group and the wallet is now under T and can neither provide a heartbeat nor the transaction to move funds.

Reply:
I'm not sure if it belongs in this RFC specifically, but I think we talked about counting pending undelegating operators as inactive for determining weak heartbeats (so they'd make the heartbeat test fail but still be able to sign the transaction to move the funds.

This means that in the scenario where, say, the undelegation time was 2 months and the max wallet time was 1 year, I think we're still okay:

```
month 1: wallet gets created with a whale that has 40% of the stake
month 6: whale undelegates
month 6: weak heartbeat detected because whale's stake is marked as inactive
month 6: underlying btc is able to get transferred to a different wallet because the whale can still sign messages
```

Reply:
That makes sense. Undelegating operators are not participating in the heartbeat but they will sign any redemption transaction. I think it is worth noting in this RFC to avoid confusion.